File: ai_studio_code.go
package main

import (
  "context"
  "flag"
  "fmt"
  "log"

  "google.golang.org/genai"
)

var model = flag.String("model", "gemini-3.1-pro-preview", "the model name, e.g. gemini-3.1-pro-preview")

func run(ctx context.Context) {
  client, err := genai.NewClient(ctx, &genai.ClientConfig{
    Backend: genai.BackendGeminiAPI,
    APIKey: os.Getenv("GEMINI_API_KEY"),
  })
  if err != nil {
    log.Fatal(err)
  }

  var tools = []*genai.Tool{
    {
      UrlContext: &genai.UrlContext{},
    },
    {
      GoogleSearch: &genai.GoogleSearch{
      },
    },
  }

  var config *genai.GenerateContentConfig = &genai.GenerateContentConfig{
    Tools: tools,
    ThinkingConfig: &genai.ThinkingConfig{
      ThinkingLevel: genai.Ptr[string]("HIGH"),
    },
  }

  var contents = []*genai.Content{
    &genai.Content{
      Role: "user",
      Parts: []*genai.Part{
        &genai.Part{
          Text: "INSERT_INPUT_HERE",
        },
      },
    },
  }

  // Call the GenerateContent method.
  result, err := client.Models.GenerateContent(ctx, *model, contents, config)
  if err != nil {
    log.Fatal(err)
  }
  fmt.Println(result.Text())
}

func main() {
  ctx := context.Background()
  flag.Parse()
  run(ctx)
}

File: ai_studio_code.java
package com.example;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableMap;
import com.google.genai.Client;
import com.google.genai.ResponseStream;
import com.google.genai.types.*;
import com.google.gson.Gson;
import com.google.genai.JsonSerializable;


import java.util.ArrayList;
import java.util.Base64;
import java.util.List;
import java.util.Map;


public class  {
  public static void main(String[] args) {
    String apiKey = System.getenv("GEMINI_API_KEY");
    Client client = Client.builder().apiKey(apiKey).build();
    Gson gson = new Gson();

    List<Tool> tools = new ArrayList<>();
    tools.add(
      Tool.builder()
        .urlContext(
          UrlContext.builder().build()
        )
        .build()
    );
    tools.add(
      Tool.builder()
        .googleSearch(
          GoogleSearch.builder()
        )
      .build());

    String model = "gemini-3.1-pro-preview";
    List<Content> contents = ImmutableList.of(
      Content.builder()
        .role("user")
        .parts(ImmutableList.of(
          Part.fromText("INSERT_INPUT_HERE")
        ))
        .build()
    );
    GenerateContentConfig config =
      GenerateContentConfig
      .builder()
      .thinkingConfig(
        ThinkingConfig
          .builder()
          .thinkingLevel("HIGH")
          .build()
      )
      .tools(tools)
      .build();

    ResponseStream<GenerateContentResponse> responseStream = client.models.generateContentStream(model, contents, config);

    for (GenerateContentResponse res : responseStream) {
      if (res.candidates().isEmpty() || res.candidates().get().get(0).content().isEmpty() || res.candidates().get().get(0).content().get().parts().isEmpty()) {
        continue;
      }

      List<Part> parts = res.candidates().get().get(0).content().get().parts().get();
      for (Part part : parts) {
        System.out.println(part.text());
      }
    }

    responseStream.close();
  }
}


